docs/DeveloperGuide.md

```diff
@@ -464,33 +464,31 @@ should end at the destroy marker (X) but due to a limitation of PlantUML, the li
 
 </div>
 
-Step 3. The user executes `show 1` to check that the specified assignment has been added for persons in the specified
-module.
-
 The following activity diagram summarizes what happens when a user executes the `give` command:
 
 <p align="center">
-  <img src="images/GiveActivityDiagram.png" width="500" height="550" />
+  <img src="images/GiveActivityDiagram.png" width="600" height="550" />
 </p>
 
 <div style="page-break-after: always;"></div>
 
 #### Design considerations
-**Aspect: Adds an assignment to a person in current displayed list or to any other person in storage:**
+**Aspect: Adds an assignment to a person in currently displayed list or to any other person in storage:**
 
-* **Alternative 1:** Add an assignment to a person in current displayed list
+* **Alternative 1 (current choice):** Add an assignment to a person in current displayed list.
     * Pros: If the displayed list is shorter, the addition of assignments will be faster.
-    * Cons: User has to carry out `list` command first if addition of assignment is desired for that person
+    * Cons: User has to carry out `list` command first if addition of assignment is desired for that person.
 
-* **Alternative 2 (current choice):** Add an assignment to any person in the storage
-    * Pros: Allows user to add assignment to a particular person even when he is not visible in the list
-    * Cons: Might take longer to execute
+* **Alternative 2:** Add an assignment to any person in the storage.
+    * Pros: Allows user to add assignment to a particular person even when he is not visible in the list.
+    * Cons: Might take longer to execute.
 
-* Considering the fact that the `give` command is meant for users to add assignments to any person in model,
-* **Alternative 2** was chosen as it meets this specification. Moreover, it will not duplicate the assignment for
+* Considering the fact that the `give` command is meant for users to add assignments to visible persons in contact list,
+* **Alternative 1** was chosen as it meets this specification. Moreover, it will not duplicate the assignment for
   persons who already have the assignment.
-  **Alternative 1** requires an additional command `list` to ensure the displayed list contains all persons before adding
-  assignments, which means it is less convenient for users as they have to do extra work.
+  **Alternative 1** requires an additional effort for user to ensure the person who he or she wants to give the assignment to
+  is actually stored in the contact list when the person is not in the currently displayed list but stored in contact list, which means mistakes 
+  about giving assignment to wrong person are likely to occur.
```

Comment:
I think you can say "...the person who the user want to give the assignment to is actually stored in the contact list but not displayed, which may lead to the user giving assignments to the wrong person"